models/workitem.go

```diff
@@ -38,3 +41,22 @@ func (wi WorkItem) Equal(u convert.Equaler) bool {
 	}
 	return wi.Fields.Equal(other.Fields)
 }
+
+// ConvertFromModel serializes a database persisted workitem.
+func (workItem WorkItem) ConvertFromModel(wiType WorkItemType) (*app.WorkItem, error) {
```

Comment:
I think this method should be on the WorkItemType, not the WorkItem. With fields, it's the "meta"-object that does the conversion, not the instance. Lets' be consistent here.